recipes/nss/all/conanfile.py

```diff
@@ -17,182 +18,182 @@ class NSSConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     description = "Network Security Services"
     topics = ("network", "security", "crypto", "ssl")
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": True, "fPIC": True}
 
+    package_type = "static-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def build_requirements(self):
-        if self.settings.compiler == "Visual Studio" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-        if self.settings.os == "Windows":
-            self.build_requires("mozilla-build/3.3")
-        if hasattr(self, "settings_build"):
-            self.build_requires("sqlite3/3.41.2")
-
     def configure(self):
-        self.options["nspr"].shared = True
```

Comment:
This change forces explicitly setting the nspr:shared=True option, even though that is required in order to build. Yes, it will fail validation below if not set, but it makes no sense to force setting the option. This problem also manifests when building Qt, as you are forced to explicitly set an option for a transitive dependency, which seems even more counterintuitive.

Reply:
Good point. Thanks! Reverted.